cmd/compare.go

```diff
@@ -94,85 +104,145 @@ func compare(provider string, oldCommit string, newCommit string) error {
 		for propName, prop := range res.Properties {
 			newProp, ok := newRes.Properties[propName]
 			if !ok {
-				violations = append(violations, fmt.Sprintf("Resource %q missing output %q", resName, propName))
+				violation("missing output %q", propName)
 				continue
 			}
 
 			vs := validateTypes(&prop.TypeSpec, &newProp.TypeSpec, fmt.Sprintf("Resource %q output %q", resName, propName))
 			violations = append(violations, vs...)
 		}
+
+		oldRequiredInputs := newSetFromList(res.RequiredInputs)
```

Comment:
Could we handle this globally in validateTypes?

Reply:
I'm not sure what you mean by "this". `validateTypes` doesn't handle `schema.ComplexTypeSpec` or `schema.ObjectTypeSpec`, which is what this pass operates on.

Reply:
This is likely a personal preference, but I would rather have a separate validation function than an inline check. Since we make `newSetFromList` handle multiple types, I was wondering if we can make `validateTypes` accept generic input and do the comparison there, since that's where I would look.

Reply:
This is operating at the `schema.*Spec` level. `res.RequiredInputs` is always `[]string`, as is `res.Required`.

`validateTypes` doesn't ever look at types with a `.Required` property, and is completely irrelevant to this pass. It operates on a `schema.TypeSpec`, which can represent simple types (`string`, `number`, `bool`, etc), collection types (`array`, `map`) and type references `$ref: ...`. It does not look inside type references.